packages/extension-chakra-storefront/src/hooks/use-toast.test.js

```diff
@@ -1,58 +1,138 @@
 /*
- * Copyright (c) 2021, salesforce.com, inc.
+ * Copyright (c) 2025, salesforce.com, inc.
  * All rights reserved.
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import React from 'react'
-import {useToast} from './use-toast'
-import {Button} from '@chakra-ui/react'
-import {renderWithProviders} from '../utils/test-utils'
-import {screen, waitFor} from '@testing-library/react'
-import userEvent from '@testing-library/user-event'
-
-jest.setTimeout(60000)
-const MockedComponent = (props) => {
-    const showToast = useToast()
-
-    const renderToast = () => {
-        showToast({
-            ...props,
-            title: 'Demo Notification',
-            status: 'success',
-            duration: 1000,
-            variant: 'subtle'
-        })
+import {renderHook} from '@testing-library/react'
+import useToast from './use-toast'
+import {toaster} from '../components/toaster'
+
+// Mock the toaster dependency
+jest.mock('../components/toaster', () => ({
+    toaster: {
+        create: jest.fn()
     }
-    return (
-        <>
-            <Button onClick={renderToast}>show toast</Button>
-        </>
-    )
-}
-
-// Set up and clean up
-beforeEach(() => {
-    jest.resetModules()
-})
+}))
+
+describe('useToast', () => {
```

Comment:
Although this is a hook, but this hook renders a visual Component. I think we should assert to check the element/text in the DOM instead of assert for arg bening passed to when this hooks is rendered.
I saw the toaster component tests also does not contain any unit tests checking UI behavior. We should have it either here or in the toaster component.  See this [comment](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2489/files#r2112891262)